FRUI/FRUI/ViewControllers/AuthStepViewController.swift

```diff
@@ -126,7 +126,33 @@ class AuthStepViewController: UIViewController {
                 //  Set auth callbacks in list for rendering
                 self.currentNode = node
                 self.authCallbacks = node.callbacks
-                self.renderAuthStep()
+                
+                //  If DeviceProfileCallback is the only Callback, then immediately collect, and submit
+                if self.authCallbacks.count == 1, let deviceProfileCallback = self.authCallbacks.first as? DeviceProfileCallback {
+                    deviceProfileCallback.execute { (profile) in
+                        self.submitNode()
+                    }
+                }
+                else {
+                    var deviceProfileCallback: DeviceProfileCallback?
+                    for callback in self.authCallbacks {
+                        if let thisCallback = callback as? DeviceProfileCallback {
+                            deviceProfileCallback = thisCallback
+                        }
```

Comment:
This block of code is mind-bending for me O_o 

Does this mean the following:

Within this conditional block, assign `callback` to `thisCallback`. Evaluate if `callback` is of type `DeviceProfileCallback`, and if it is, assign the value of `thisCallback` (which was `callback`) to the variable `deviceProfileCallback`.

If the above is true, should I assume conditionals in Swift require a `let` statement to assign the value being evaluated to something? Sorry, if that's a dumb question :)

Reply:
@cerebrl , this is more of Swift syntax, and it maybe difficult to understand as we haven't seen such thing in other language. So, let me explain line by line.

So, in line#137, I declare the optional variable named, deviceProfileCallback which is currently null. 
In line#138, I'm looping through list of Callbacks in Node. 
In line#139, I'm checking if the callback in the list is DeviceProfileCallback; this may look strange, but as Node contains an array of Callback (which is super class of all Callback), I need to typecast it to DeviceProfileCallback, and that's essentially what this if statement does. You can also read it as "if this callback is DeviceProfileCallback".
In line#140, I'm assigning the DeviceProfileCallback to deviceProfileCallback variable that I declared in line#137. The reason that I keep it as variable is similar to what I answered above in the other comment. As DeviceProfile collecting operation is async, in iOS, I try to avoid running this in UI thread (and showing loading indicator when it runs in background), and ensure that the operation is completed before I render UI. So, I'm using this variable later part of the codes to collect DeviceProfile.

In short, Swift requires let statement to assign the value being evaluated to something else.  This let statement unwraps the variable and/or typcast the variable as needed. Please let me know if you have any other question. I would love to discuss about it if you want.

Reply:
Thanks for the help understanding the code, @jhoongo! I think I get it. I'll let this settle in my mind for a bit :)